test/pfm/pfm_test.go

```diff
@@ -0,0 +1,237 @@
+	Forward *ForwardMetadata `json:"forward"`
+}
+
+type ForwardMetadata struct {
+	Receiver       string        `json:"receiver"`
+	Port           string        `json:"port"`
+	Channel        string        `json:"channel"`
+	Timeout        time.Duration `json:"timeout"`
+	Retries        *uint8        `json:"retries,omitempty"`
+	Next           *string       `json:"next,omitempty"`
+	RefundSequence *uint64       `json:"refund_sequence,omitempty"`
+}
+
+func SetupTest(t *testing.T) (*ibctesting.Coordinator, *ibctesting.TestChain,
+	*ibctesting.TestChain, *ibctesting.TestChain,
+) {
+	chains := make(map[string]*ibctesting.TestChain)
+	coordinator := &ibctesting.Coordinator{
+		T:           t,
+		CurrentTime: time.Now(),
+		Chains:      chains,
+	}
+	celestiaChain := utils.NewTestChain(t, coordinator, ibctesting.GetChainID(1))
+	chainA := NewTestChain(t, coordinator, ibctesting.GetChainID(2))
+	chainB := NewTestChain(t, coordinator, ibctesting.GetChainID(3))
+	coordinator.Chains[ibctesting.GetChainID(1)] = celestiaChain
+	coordinator.Chains[ibctesting.GetChainID(2)] = chainA
+	coordinator.Chains[ibctesting.GetChainID(3)] = chainB
+	return coordinator, chainA, celestiaChain, chainB
+}
+
+func NewTransferPaths(chain1, chain2, chain3 *ibctesting.TestChain) (*ibctesting.Path, *ibctesting.Path) {
+	path1 := ibctesting.NewPath(chain1, chain2)
+	path1.EndpointA.ChannelConfig.PortID = ibctesting.TransferPort
+	path1.EndpointB.ChannelConfig.PortID = ibctesting.TransferPort
+	path1.EndpointA.ChannelConfig.Version = types.Version
+	path1.EndpointB.ChannelConfig.Version = types.Version
+	path2 := ibctesting.NewPath(chain2, chain3)
+	path2.EndpointA.ChannelConfig.PortID = ibctesting.TransferPort
+	path2.EndpointB.ChannelConfig.PortID = ibctesting.TransferPort
+	path2.EndpointA.ChannelConfig.Version = types.Version
+	path2.EndpointB.ChannelConfig.Version = types.Version
+
+	return path1, path2
+}
+
+// TestPacketForwardMiddlewareTransfer sends a PFM transfer originating from Celestia to ChainA, then back to Celestia and finally to ChainB.
+// It verifies that Celestia forwards the packet successfully, the balance of the sender account on Celestia decreases by the amount sent,
+// and the balance of the receiver account on ChainB increases by the amount sent.
+func TestPacketForwardMiddlewareTransfer(t *testing.T) {
+	coordinator, chainA, celestia, chainB := SetupTest(t)
+	path1, path2 := NewTransferPaths(chainA, celestia, chainB)
+
+	coordinator.Setup(path1)
+	coordinator.Setup(path2)
+
+	celestiaApp := celestia.App.(*app.App)
+	originalCelestiaBalalance := celestiaApp.BankKeeper.GetBalance(celestia.GetContext(), celestia.SenderAccount.GetAddress(), sdk.DefaultBondDenom)
+
+	// Take half of the original balance
+	transferAmount := originalCelestiaBalalance.Amount.QuoRaw(2)
+	timeoutHeight := clienttypes.NewHeight(1, 300)
+	coinToSendToB := sdk.NewCoin(sdk.DefaultBondDenom, transferAmount)
+
+	// Forward the packet to ChainB
+	secondHopMetaData := &PacketMetadata{
```

Comment:
IMO "D" does not need to be capitalized because it is "metadata"

secondHopMetaData => secondHopMetadata